spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNetConnection.java

```diff
@@ -125,9 +125,16 @@ public synchronized void send(Message<?> message) {
 
 	@Override
 	public Object getPayload() {
+		InputStream inputStream;
+		try {
+			inputStream = inputStream();
+		}
+		catch (IOException e1) {
+			throw new UncheckedIOException(new SoftEndOfStreamException("Socket closed when getting input stream", e1));
```

Comment:
Can we make a `SoftEndOfStreamException` as an `UncheckedIOException` to avoid this wrapping?

i don't mind for other branches, but at least for the `master`.

Thanks